low-level/sensors/impl/generic/node_hw.py

```diff
@@ -480,6 +481,54 @@ def run(self):
             logger.warning(f"{self.SENSOR_NAME} Node hw monitoring disabled")
             self.shutdown()
 
+    def _check_missmatch_resource_status(self):
+        for sensor_id in self.faulty_resources:
+            if self.faulty_resources[sensor_id]['resource'] in ['fan', 'disk', 'temperature', 'voltage']:
+                _, sensor_get_data, _ = self._get_sensor_props(sensor_id)
+                if sensor_get_data and 'Assertion Events' in sensor_get_data \
+                        and 'Status' in sensor_get_data:
+                    # 'Status': 'ok'
+                    # 'Assertion Events': ''
+                    if sensor_get_data['Assertion Events'] in [""] \
+                        and sensor_get_data['Status'] == 'ok':
+                        self._create_sel_event_entry(sensor_id)
+
+            else:
+                _, sdr_get_data = self._get_sensor_sdr_props(sensor_id)
+                if sdr_get_data and 'States Asserted' in sdr_get_data:
+                    #  'States Asserted': 'Power Supply, Presence detected'
+                    resource_state = re.sub(',  +', ', ', re.sub('[\[\]]','',
+                        sdr_get_data['States Asserted']).replace('\n',','))
+                    sensor_status = resource_state.split(',')
+                    sensor_status = sensor_status[0] if len(sensor_status) == 1 else sensor_status[1]
+                    sensor_status = sensor_status.strip()
+                    if sensor_status in ["", "Presence detected"]:
+                        self._create_sel_event_entry(sensor_id)
+
+    def _create_sel_event_entry(self, sensor_id):
+        event_map_dict = {
+            'Lower Non-Recoverable': 'lnr',
+            'Lower Critical': 'lcr',
+            'Lower Non-Critical': 'lnc',
+            'Upper Non-Critical': 'unc',
+            'Upper Critical': 'ucr',
+            'Upper Non-Recoverable': 'unr'
+        }
+        event = self.faulty_resources[sensor_id]['event']
+        if "going high" in event.lower() or "going low" in event.lower():
+            event = next(v for k,v in event_map_dict.items() if event.startswith(k))
+        # "Presence detected ()"
+        event = event.replace('()', '').strip()
+        status = self.faulty_resources[sensor_id]['status']
+        status = 'deasserted' if status == 'asserted' else 'asserted'
+        logger.warn(f"Found missing entry in the sel list for %s resource, \
+            Updating the sel list manually" % (sensor_id))
+        sel_event, retcode = self._run_ipmitool_subcommand("event '%s' '%s' %s" % (
```

Comment:
Ensure Fan events are running for fan sensor and PSU events are running for PSU sensor and also Disk/Drive with its respective event.

Reply:
Yes. That would be taken care by "self.faulty_resources" varibale.